lib/src/backend.rs

```diff
@@ -137,6 +137,18 @@ content_hash! {
     }
 }
 
+impl Signature {
+    pub fn username(&self) -> Option<&str> {
+        // While this is not perfect (i.e., not exactly RFC 5322 compatible), it
+        // is very simple and should work well for a vast majority of the cases.
+        let parts: Vec<&str> = self.email.split('@').collect();
```

Comment:
nit: use `splitn(1, '@')` to avoid splitting parts we don't care about (though they're unlikely to exist in practice).

Reply:
`split_once('@')` is even better.

FWIW, I would add `split_email(str) -> (str, Option<str>)` to template_parser/templater since it's unlikely we would need such function in lib.

Reply:
I also thought about the backend vs. templater location but I wasn't sure we might want the same definition of "username" in a future GUI. But maybe it's best to put it in the templater for now and then move it into the library if we see a use for it?

Reply:
This was intentional -- IIUC, the spec does allow for multiple @ but if they're in quotes or something. By returning None here for cases we don't "understand" well, we just display the whole email in template. WDYT? (Not tied to this and happy to change to splitn though)

Reply:
Sorry, I had a draft reply to Martin's original comment and didn't see the new replies till now, will move to templater!

Reply:
> I wasn't sure we might want the same definition of "username" in a future GUI. 

I think this function belongs to the category of `time_util.rs` (ignoring the "time" part.) It'll be eventually extracted to crate/module, but we don't have the exact place atm.